wrappers/kv_quorum.go

```diff
@@ -0,0 +1,385 @@
+package wrappers
+
+import (
+	"sync/atomic"
+	"time"
+
+	"github.com/portworx/kvdb"
+	"github.com/sirupsen/logrus"
+)
+
+type kvQuorumCheckFilter struct {
+	// kv is the kvdb which is wrapped
+	kv kvdb.Kvdb
+	// quorumState is kvdb quorum state
+	quorumState uint32
+}
+
+// NewKvQuorumCheckFilter constructs a new kvdb.Kvdb.
+func NewKvQuorumCheckFilter(
+	kv kvdb.Kvdb,
+	domain string,
+	machines []string,
+	options map[string]string,
+	fatalErrorCb kvdb.FatalErrorCB,
+) (kvdb.Kvdb, error) {
+	logrus.Infof("Registering with retry wrapper")
+	return &kvQuorumCheckFilter{
+		kv:          kv,
+		quorumState: uint32(kvdb.KvdbInQuorum),
+	}, nil
+}
+
+func (k *kvQuorumCheckFilter) SetQuorumState(state kvdb.KvdbQuorumState) {
```

Comment:
When and who is going to invoke this?

Reply:
the idea is that clients using kvdb package will use watch errors to indicate kvdb quorum state, that way we keep most of current code in each package untouched.